impeller/toolkit/glvk/trampoline.cc

```diff
@@ -0,0 +1,328 @@
+                          0                      //
+  );
+
+  FML_CHECK(gl.CheckFramebufferStatus(GL_FRAMEBUFFER) ==
+            GL_FRAMEBUFFER_COMPLETE);
+
+  gl.Disable(GL_BLEND);
+  gl.Disable(GL_SCISSOR_TEST);
+  gl.Disable(GL_DITHER);
+  gl.Disable(GL_CULL_FACE);
+  gl.ColorMask(GL_TRUE, GL_TRUE, GL_TRUE, GL_TRUE);
+
+  gl.ClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+  gl.Clear(GL_COLOR_BUFFER_BIT);
+
+  const auto& fb_size = dst_texture.GetTextureDescriptor().size;
+  gl.Viewport(0, 0, fb_size.width, fb_size.height);
+
+  gl.UseProgram(program_);
+
+  struct VertexData {
+    Point position;
+    Point tex_coord;
+  };
+
+  // The vertex coordinates assume OpenGL NDC because that the API we are using
+  // to draw the quad. But the texture will be sampled in Vulkan so the texture
+  // coordinate system assumes Vulkan convention.
+  //
+  // See the following help link for an overview of the different coordinate
+  // systems:
+  // https://github.com/flutter/engine/blob/5810b3fc791f4bb82b9a454014310990eddc1181/impeller/docs/coordinate_system.md
+  static constexpr const VertexData kVertData[] = {
+      {{-1, -1}, {0, 1}},  // bottom left
+      {{-1, +1}, {0, 0}},  // top left
+      {{+1, +1}, {1, 0}},  // top right
+      {{+1, -1}, {1, 1}},  // bottom right
+  };
+
+  // This is tedious but we assume no vertex array objects (VAO) are available
+  // because of ES 2 versioning constraints.
+  GLuint vertex_buffer = GL_NONE;
+  gl.GenBuffers(1u, &vertex_buffer);
+  gl.BindBuffer(GL_ARRAY_BUFFER, vertex_buffer);
+  gl.BufferData(GL_ARRAY_BUFFER, sizeof(kVertData), kVertData, GL_STATIC_DRAW);
+  gl.EnableVertexAttribArray(kAttributeIndexPosition);
+  gl.EnableVertexAttribArray(kAttributeIndexTexCoord);
+  gl.VertexAttribPointer(kAttributeIndexPosition, 2, GL_FLOAT, GL_FALSE,
+                         sizeof(VertexData),
+                         (void*)offsetof(VertexData, position));
+  gl.VertexAttribPointer(kAttributeIndexTexCoord, 2, GL_FLOAT, GL_FALSE,
+                         sizeof(VertexData),
+                         (void*)offsetof(VertexData, tex_coord));
+
+  gl.ActiveTexture(GL_TEXTURE0);
+  gl.BindTexture(src_texture.target, src_texture.texture);
+  gl.Uniform1i(texture_uniform_location_, 0u);
+
+  auto gl_st_transformation = src_texture.st_transformation;
+
+  gl.UniformMatrix4fv(st_transformation_location_, 1u, GL_FALSE,
+                      reinterpret_cast<GLfloat*>(&gl_st_transformation));
+
+  gl.DrawArrays(GL_TRIANGLE_FAN, 0, 4);
+
+  gl.UseProgram(GL_NONE);
+
+  gl.Flush();
+
+  gl.DeleteFramebuffers(1u, &offscreen_fbo);
+  gl.DeleteTextures(1u, &dst_gl_texture);
+  gl.DeleteBuffers(1u, &vertex_buffer);
+
+  // Theoretically, this does nothing because the surface is a 1x1 pbuffer
+  // surface. But frame capture tools use this to denote a frame boundary in
+  // OpenGL. So add this as a debugging aid anyway.
+  eglSwapBuffers(egl_display_->GetHandle(), egl_surface_->GetHandle());
```

Comment:
Back when I was trying to debug texture coordinates, I wanted to see if my render-to-texture was bad in OpenGL or the sampling was in Vulkan. Mali Graphics Analyzer only allows displaying the contents of textures and intermediate framebuffers at frame boundaries. And while rendering to a texture in OpenGL, without an eglSwapBuffers, the tool thought that the first frame was never done because there was no swap. So thousands of commands kept getting captured as being part of the "frame". The tool was just waiting indefinitely.

Adding this innocuous call made the captures work and nicely separated the OpenGL workloads from the Vulkan ones. But, in terms of actual work done, this is a no-op.

<img width="3120" alt="debugging" src="https://github.com/user-attachments/assets/424eef6c-5e89-44a8-9160-72e7047585c6">